### DISCUSSION TITLE
Ошибка компиляции

### DISCUSSION THREAD

**Discussion by sepulka123** [vvip-68/LedPanelWiFi#26] [Вопросы и ответы]
Добрый день.
Что то не получается у меня прошить версию 1.14 в контроллер ESP-32.
[txt.txt](https://github.com/vvip-68/LedPanelWiFi/files/13505427/txt.txt)
Вроде и ядро правильное и библиотеки переписал из архива.
Я уже недавно прошивал эту версию. Потом захотел вернуться к версии 1.13.
А вот теперь снова 1.14, и никак не хочет компилироваться.
В текстовом файлике есть краткий и подробный отчет при компиляции.
Может что напутал с библиотеками. Переустанавливал "Ардуино". Компилятор пишет, что 
несколько библиотек "...эта используется, эта не используется".
Может что посоветуете, в какую сторону копать, где посмотреть ?

**Comment by vvip-68** [comment]
Единственное, что могу сказать - оно ругается на библиотеку Time.
Или не той версии или еще что-то.
Попробуйте сначала просто удалить папку Time из папки библиотек  которая в "Program Files (x86)/Arduino/libraries", также если есть - в Documents and Settings/<user>/Documents/Arduino/libraries.
Если после этого не скомпилится - снова скопируйте из папки проекта libraries в папку "Program Files (x86)/Arduino/libraries"  папку Time и пересоберите проект...

**Comment by sepulka123** [comment]
Удалил "Time" из Ардуино. Записал наново из архива проекта.
Скомпилировалось и прошилось - без проблем !
Но при попытке записи Веб-страницы - выдает ошибку:
Arduino: 1.8.19 (Windows 10), Плата:"ESP32 Dev Module, Disabled, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), 240MHz (WiFi/BT), QIO, 80MHz, 4MB (32Mb), 921600, None"



LITTLEFS Error: mklittlefs not found!
"esp32littlefs.jar" записал в библиотеку !

И самое обидное, что я же это уже делал в мае месяце.
И все прекрасно прошло. Где же все оно делось ?
Правда это была другая версия ядра (более старшая, наверное самая новая, теперь уже не вспомнить).
(переустанавливал Ардуино)
Может туда входила и "mklittlefs".
Нашел в интернете "mklittlefs_esp32-main.zip" , разархивировал в библиотеку.
Но ошибка та же.
[mklittlefs_esp32-main.zip](https://github.com/vvip-68/LedPanelWiFi/files/13514176/mklittlefs_esp32-main.zip)

**Reply by vvip-68** [reply]
А это от ядра не зависит. Запись в файловую систему выполняется отдельным плагином, который должен быть установлен в IDE. Сам плагин - в папке tools проекта: LittleFS_Uploader. 

А что значит '"esp32littlefs.jar" записал в библиотеку !'? В какую библиотеку? При чем тут библиотека?

https://github.com/vvip-68/LedPanelWiFi/wiki/Загрузка-данных-в-файловую-систему-МК

**Comment by sepulka123** [comment]
Да-да. Именно в папке "Tools" Ардуино.
Папка Tools в моем Ардуино теперь выглядит так:
[tools.zip](https://github.com/vvip-68/LedPanelWiFi/files/13517157/tools.zip)
Получилось в нескольких вариантах: просто в папке, в подпапке.
Перезаписывал и каждый раз перезагружал Ардуино прежде чем нажимать кнопку "Data Upload".
Результат неизменный: 
LITTLEFS Error: mklittlefs not found!

**Comment by sepulka123** [comment]
Похоже, что нашел !
Несмотря на то, что я использую контроллер ESP-32, обязательно необходимо устанавливать ядро 8266 !
Установил - и все записалось (во всяком случае ошибки нет и что то там загружалось) !
То ли оно, что нужно - буду проверять уже завтра.
Благодарю за помощь !

**Reply by vvip-68** [reply]
Ну, для меня это новость. И логической связи я не вижу. Ну, пусть будет так.
У меня то оба ядра всегда были установлены...

**Comment by sepulka123** [comment]
Без проблем !
Благодарю !
![P31201-162904](https://github.com/vvip-68/LedPanelWiFi/assets/55927434/03857c18-2d8f-46bc-ab4d-b2abb093efd9)
